File: backend/api/src/Models/ShopDb/PasswordResetToken.cs
using System;
using System.Text.Json.Serialization;

namespace Models.ShopDb;

public partial class PasswordResetToken
{
    public long Id { get; set; }

    public long UserId { get; set; }

    public Guid Token { get; set; }

    public DateTime ExpiresAt { get; set; }

    public DateTime? ConsumedAt { get; set; }

    public bool IsRevoked { get; set; }

    public DateTime CreatedAt { get; set; }

    [JsonIgnore] public virtual User User { get; set; } = null!;
}

File: backend/sso/internal/domain/password_reset_token.go
package domain

import "time"

type PasswordResetToken struct {
	Id         int64      `db:"id"`
	UserId     int64      `db:"user_id"`
	Token      string     `db:"token"`
	ExpiresAt  time.Time  `db:"expires_at"`
	ConsumedAt *time.Time `db:"consumed_at"`
	IsRevoked  bool       `db:"is_revoked"`
	CreatedAt  time.Time  `db:"created_at"`
}
